### ISSUE TITLE
Example won't compile

### ISSUE THREAD

**Issue by robzsaunders** [tdlight-team/tdlight-java#186]
First off, when trying to pull the dependencies I get these errors.

```
[WARNING] Failed to download log4j-core-2.20.0.pom [https://mvn.mchv.eu/repository/mchv/]
[WARNING] Failed to download log4j-slf4j2-impl-2.20.0.pom [https://mvn.mchv.eu/repository/mchv/]
[WARNING] Failed to download slf4j-api-2.0.6.pom [https://mvn.mchv.eu/repository/mchv/]
[WARNING] Failed to download slf4j-parent-2.0.6.pom [https://mvn.mchv.eu/repository/mchv/]
[WARNING] Failed to download disruptor-3.4.4.pom [https://mvn.mchv.eu/repository/mchv/]
[WARNING] Failed to download slf4j-api-2.0.6.jar [https://mvn.mchv.eu/repository/mchv/]
[WARNING] Failed to download disruptor-3.4.4.jar [https://mvn.mchv.eu/repository/mchv/]
[WARNING] Failed to download log4j-slf4j2-impl-2.20.0.jar [https://mvn.mchv.eu/repository/mchv/]
[WARNING] Failed to download log4j-core-2.20.0.jar [https://mvn.mchv.eu/repository/mchv/]
```

Second, it looks like whenever this error comes up you have fixed it with a new release? Can you take a look please.

```
Exception in thread "main" it.tdlight.util.UnsupportedNativeLibraryException: Failed to load TDLight native libraries
	at it.tdlight.util.Native.loadLibrary(Native.java:65)
	at it.tdlight.util.Native.loadNativesInternal(Native.java:37)
	at it.tdlight.Init.init(Init.java:53)
	at org.myorg.Example.main(Example.java:37)
Caused by: java.lang.IllegalArgumentException: Failed to load any of the given libraries: [tdjni.windows_amd64]
	at it.tdlight.util.NativeLibraryLoader.loadFirstAvailable(NativeLibraryLoader.java:132)
	at it.tdlight.util.Native.loadLibrary(Native.java:55)
	... 3 more
	Suppressed: java.lang.UnsatisfiedLinkError: could not load a native library: tdjni.windows_amd64
		at it.tdlight.util.NativeLibraryLoader.load(NativeLibraryLoader.java:257)
		at it.tdlight.util.NativeLibraryLoader.loadFirstAvailable(NativeLibraryLoader.java:123)
		... 4 more
	Caused by: java.io.FileNotFoundException: META-INF/tdlightjni/tdjni.windows_amd64.dll
		at it.tdlight.util.NativeLibraryLoader.load(NativeLibraryLoader.java:204)
		... 5 more
		Suppressed: java.lang.UnsatisfiedLinkError: no tdjni.windows_amd64 in java.library.path: C:\Program Files\Java\jdk-17.0.4.1\bin;C:\WINDOWS\Sun\Java\bin;C:\WINDOWS\system32;C:\WINDOWS;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\bin;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\libnvvp;C:\Program Files\NetBeans-12.2\netbeans\java\maven\\bin;C:\Program Files (x86)\Gpg4win\..\GnuPG\bin;C:\Users\dulak\AppData\Local\Programs\Python\Python311;C:\Users\dulak\AppData\Local\Programs\Python\Python311\Scripts;C:\Program Files\CMake\bin;C:\Program Files\NVIDIA Corporation\Nsight Compute 2023.2.2\;C:\Program Files\Docker\Docker\resources\bin;C:\WINDOWS\system32\config\systemprofile\AppData\Roaming\pypoetry\venv\Scripts;C:\Users\dulak\AppData\Roaming\pypoetry\venv\Scripts;C:\Users\admin;C:\Users\dulak\AppData\Roaming\Python\Scripts;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\bin;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\libnvvp;C:\Program Files\NetBeans-12.2\netbeans\java\maven\\bin;C:\Program Files (x86)\Gpg4win\..\GnuPG\bin;C:\Users\dulak\AppData\Local\Programs\Python\Python311;C:\Program Files\CMake\bin;C:\Program Files\NVIDIA Corporation\Nsight Compute 2023.2.2\;C:\Program Files\Docker\Docker\resources\bin;C:\Users\dulak\AppData\Local\Microsoft\WindowsApps;;C:\Program Files\JetBrains\RustRover 233.10527.39\bin;;.
			at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2429)
			at java.base/java.lang.Runtime.loadLibrary0(Runtime.java:818)
			at java.base/java.lang.System.loadLibrary(System.java:1989)
			at it.tdlight.util.NativeLibraryUtil.loadLibrary(NativeLibraryUtil.java:23)
			at it.tdlight.util.NativeLibraryLoader.loadLibrary(NativeLibraryLoader.java:439)
			at it.tdlight.util.NativeLibraryLoader.load(NativeLibraryLoader.java:179)
			... 5 more
			Suppressed: java.lang.UnsatisfiedLinkError: no tdjni.windows_amd64 in java.library.path: C:\Program Files\Java\jdk-17.0.4.1\bin;C:\WINDOWS\Sun\Java\bin;C:\WINDOWS\system32;C:\WINDOWS;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\bin;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\libnvvp;C:\Program Files\NetBeans-12.2\netbeans\java\maven\\bin;C:\Program Files (x86)\Gpg4win\..\GnuPG\bin;C:\Users\dulak\AppData\Local\Programs\Python\Python311;C:\Users\dulak\AppData\Local\Programs\Python\Python311\Scripts;C:\Program Files\CMake\bin;C:\Program Files\NVIDIA Corporation\Nsight Compute 2023.2.2\;C:\Program Files\Docker\Docker\resources\bin;C:\WINDOWS\system32\config\systemprofile\AppData\Roaming\pypoetry\venv\Scripts;C:\Users\dulak\AppData\Roaming\pypoetry\venv\Scripts;C:\Users\admin;C:\Users\dulak\AppData\Roaming\Python\Scripts;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\bin;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\libnvvp;C:\Program Files\NetBeans-12.2\netbeans\java\maven\\bin;C:\Program Files (x86)\Gpg4win\..\GnuPG\bin;C:\Users\dulak\AppData\Local\Programs\Python\Python311;C:\Program Files\CMake\bin;C:\Program Files\NVIDIA Corporation\Nsight Compute 2023.2.2\;C:\Program Files\Docker\Docker\resources\bin;C:\Users\dulak\AppData\Local\Microsoft\WindowsApps;;C:\Program Files\JetBrains\RustRover 233.10527.39\bin;;.
				at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2429)
				at java.base/java.lang.Runtime.loadLibrary0(Runtime.java:818)
				at java.base/java.lang.System.loadLibrary(System.java:1989)
				at it.tdlight.util.NativeLibraryUtil.loadLibrary(NativeLibraryUtil.java:23)
				at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
				at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
				at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
				at java.base/java.lang.reflect.Method.invoke(Method.java:568)
				at it.tdlight.util.NativeLibraryLoader.loadLibraryByHelper(NativeLibraryLoader.java:468)
				at it.tdlight.util.NativeLibraryLoader.loadLibrary(NativeLibraryLoader.java:431)
				... 6 more
	Suppressed: java.lang.UnsatisfiedLinkError: could not load a native library: tdjni
		at it.tdlight.util.NativeLibraryLoader.load(NativeLibraryLoader.java:257)
		at it.tdlight.util.Native.loadLibrary(Native.java:58)
		... 3 more
	Caused by: java.io.FileNotFoundException: META-INF/tdlightjni/tdjni.dll
		at it.tdlight.util.NativeLibraryLoader.load(NativeLibraryLoader.java:204)
		... 4 more
		Suppressed: java.lang.UnsatisfiedLinkError: no tdjni in java.library.path: C:\Program Files\Java\jdk-17.0.4.1\bin;C:\WINDOWS\Sun\Java\bin;C:\WINDOWS\system32;C:\WINDOWS;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\bin;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\libnvvp;C:\Program Files\NetBeans-12.2\netbeans\java\maven\\bin;C:\Program Files (x86)\Gpg4win\..\GnuPG\bin;C:\Users\dulak\AppData\Local\Programs\Python\Python311;C:\Users\dulak\AppData\Local\Programs\Python\Python311\Scripts;C:\Program Files\CMake\bin;C:\Program Files\NVIDIA Corporation\Nsight Compute 2023.2.2\;C:\Program Files\Docker\Docker\resources\bin;C:\WINDOWS\system32\config\systemprofile\AppData\Roaming\pypoetry\venv\Scripts;C:\Users\dulak\AppData\Roaming\pypoetry\venv\Scripts;C:\Users\admin;C:\Users\dulak\AppData\Roaming\Python\Scripts;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\bin;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\libnvvp;C:\Program Files\NetBeans-12.2\netbeans\java\maven\\bin;C:\Program Files (x86)\Gpg4win\..\GnuPG\bin;C:\Users\dulak\AppData\Local\Programs\Python\Python311;C:\Program Files\CMake\bin;C:\Program Files\NVIDIA Corporation\Nsight Compute 2023.2.2\;C:\Program Files\Docker\Docker\resources\bin;C:\Users\dulak\AppData\Local\Microsoft\WindowsApps;;C:\Program Files\JetBrains\RustRover 233.10527.39\bin;;.
			at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2429)
			at java.base/java.lang.Runtime.loadLibrary0(Runtime.java:818)
			at java.base/java.lang.System.loadLibrary(System.java:1989)
			at it.tdlight.util.NativeLibraryUtil.loadLibrary(NativeLibraryUtil.java:23)
			at it.tdlight.util.NativeLibraryLoader.loadLibrary(NativeLibraryLoader.java:439)
			at it.tdlight.util.NativeLibraryLoader.load(NativeLibraryLoader.java:179)
			... 4 more
			Suppressed: java.lang.UnsatisfiedLinkError: no tdjni in java.library.path: C:\Program Files\Java\jdk-17.0.4.1\bin;C:\WINDOWS\Sun\Java\bin;C:\WINDOWS\system32;C:\WINDOWS;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\bin;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\libnvvp;C:\Program Files\NetBeans-12.2\netbeans\java\maven\\bin;C:\Program Files (x86)\Gpg4win\..\GnuPG\bin;C:\Users\dulak\AppData\Local\Programs\Python\Python311;C:\Users\dulak\AppData\Local\Programs\Python\Python311\Scripts;C:\Program Files\CMake\bin;C:\Program Files\NVIDIA Corporation\Nsight Compute 2023.2.2\;C:\Program Files\Docker\Docker\resources\bin;C:\WINDOWS\system32\config\systemprofile\AppData\Roaming\pypoetry\venv\Scripts;C:\Users\dulak\AppData\Roaming\pypoetry\venv\Scripts;C:\Users\admin;C:\Users\dulak\AppData\Roaming\Python\Scripts;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\bin;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.2\libnvvp;C:\Program Files\NetBeans-12.2\netbeans\java\maven\\bin;C:\Program Files (x86)\Gpg4win\..\GnuPG\bin;C:\Users\dulak\AppData\Local\Programs\Python\Python311;C:\Program Files\CMake\bin;C:\Program Files\NVIDIA Corporation\Nsight Compute 2023.2.2\;C:\Program Files\Docker\Docker\resources\bin;C:\Users\dulak\AppData\Local\Microsoft\WindowsApps;;C:\Program Files\JetBrains\RustRover 233.10527.39\bin;;.
				at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2429)
				at java.base/java.lang.Runtime.loadLibrary0(Runtime.java:818)
				at java.base/java.lang.System.loadLibrary(System.java:1989)
				at it.tdlight.util.NativeLibraryUtil.loadLibrary(NativeLibraryUtil.java:23)
				at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
				at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
				at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
				at java.base/java.lang.reflect.Method.invoke(Method.java:568)
				at it.tdlight.util.NativeLibraryLoader.loadLibraryByHelper(NativeLibraryLoader.java:468)
				at it.tdlight.util.NativeLibraryLoader.loadLibrary(NativeLibraryLoader.java:431)
				... 5 more

Process finished with exit code 1
```

**Comment by cavallium** [comment]
Try again, the repository has been offline.
By the way, did you add the windows natives?

**Comment by robzsaunders** [comment]
Hey there, thanks for getting back to me quickly.

Repos look to be working now, thanks on that side but I'm still getting the native errors.

I added this
```
        <dependency>
            <groupId>it.tdlight</groupId>
            <artifactId>tdlight-natives</artifactId>
            <classifier>windows_amd64</classifier>
        </dependency>
```

Forgive my noobie question but is that what you mean?

**Comment by robzsaunders** [comment]
the windows .jar is only 3kb big and has nothing in it compared to some of the linux .jars and the macos .jar

Im not sure how you have it setup but maybe you accidentally didn't properly compile the windows jar? Or maybe mine didn't download right?

![image](https://github.com/tdlight-team/tdlight-java/assets/78615154/925d774d-b6a2-429f-95b8-7943ef1e0263)

![image](https://github.com/tdlight-team/tdlight-java/assets/78615154/f43dfefd-fc58-47dd-be78-6d30556b6235)

**Comment by robzsaunders** [comment]
Yea I downloaded an old version and it has a actual .dll file in the windows release.

**Comment by Sascha-T** [comment]
I'm also currently having this issue (no natives on `windows_amd64`, jar only being 3 KB).

Perhaps this should be re-opened?

**Comment by Sascha-T** [comment]
> I'm also currently having this issue (no natives on `windows_amd64`, jar only being 3 KB).
> 
> Perhaps this should be re-opened?

Affected versions: `3.2.1+td.1.8.21`, `3.2.0+td.1.8.20`

**Comment by robzsaunders** [comment]
Yea not perhaps, it needs to be re-opened lol

He closed it a bit too early assuming it was the repo's :P

**Comment by cavallium** [comment]
There was a bug in the windows build pipeline, I'm rebuilding it alongside the new tdlib version, sorry for the inconvenience.

**Comment by cavallium** [comment]
There is a global Docker.io outage, I will rebuild it later :man_shrugging:

**Comment by cavallium** [comment]
https://www.dockerstatus.com

**Comment by cavallium** [comment]
Fixed in [v3.2.2+td.1.8.21](https://github.com/tdlight-team/tdlight-java/tree/v3.2.2%2Btd.1.8.21)

**Comment by robzsaunders** [comment]
Hey @cavallium thanks for promptly fixing this! Appreciate it!
